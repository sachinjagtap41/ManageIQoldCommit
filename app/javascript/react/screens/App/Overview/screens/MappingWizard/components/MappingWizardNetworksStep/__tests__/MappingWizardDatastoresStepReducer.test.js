import Immutable from 'seamless-immutable';

import networksReducer from '../MappingWizardNetworksStepReducer';
import {
  FETCH_V2V_SOURCE_NETWORKS,
  FETCH_V2V_TARGET_NETWORKS
} from '../MappingWizardNetworksStepConstants';

import {
  sourceClusterNetworks,
  targetClusterNetworks
} from '../mappingWizardNetworksStep.fixtures';

const initialState = Immutable({
  isFetchingSourceNetworks: false,
  isRejectedSourceNetworks: false,
  errorSourceNetworks: null,
  isFetchingTargetNetworks: false,
  isRejectedTargetNetworks: false,
  errorTargetNetworks: null,
  sourceNetworks: [],
  targetNetworks: []
});

test('sets default state', () => {
  const action = { type: '@@INIT' };
  const state = networksReducer(undefined, action);

  expect(state).toMatchSnapshot();
});

describe('fetching source networks', () => {
  test('is pending', () => {
    const action = {
      type: `${FETCH_V2V_SOURCE_NETWORKS}_PENDING`
    };
    const prevState = initialState.set('isRejectedSourceNetworks', true);
    const state = networksReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  test('is rejected', () => {
    const action = {
      type: `${FETCH_V2V_SOURCE_NETWORKS}_REJECTED`,
      payload: 'error'
    };
    const prevState = initialState.set('isFetchingSourceNetworks', true);
    const state = networksReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  test('is fulfilled', () => {
    const payload = {
      sourceNetworks: [sourceClusterNetworks[0].lans[0]]
    };
    const action = {
      type: `${FETCH_V2V_SOURCE_NETWORKS}_FULFILLED`,
      payload
    };
    const prevState = initialState
      .set('isRejectedSourceNetworks', true)
      .set('isFetchingSourceNetworks', true);
    const state = networksReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });
});

describe('fetching target networks', () => {
  test('is pending', () => {
    const action = {
      type: `${FETCH_V2V_TARGET_NETWORKS}_PENDING`
    };
    const prevState = initialState.set('isRejectedTargetNetworks', true);
    const state = networksReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  test('is rejected', () => {
    const action = {
      type: `${FETCH_V2V_TARGET_NETWORKS}_REJECTED`,
      payload: 'error'
    };
    const prevState = initialState.set('isFetchingTargetNetworks', true);
    const state = networksReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  test('is fulfilled', () => {
    const payload = {
      targetNetworks: [targetClusterNetworks[0].lans[0]]
    };
    const action = {
      type: `${FETCH_V2V_TARGET_NETWORKS}_FULFILLED`,
      payload
    };
    const prevState = initialState
      .set('isRejectedTargetNetworks', true)
      .set('isFetchingTargetNetworks', true);
    const state = networksReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });
});
