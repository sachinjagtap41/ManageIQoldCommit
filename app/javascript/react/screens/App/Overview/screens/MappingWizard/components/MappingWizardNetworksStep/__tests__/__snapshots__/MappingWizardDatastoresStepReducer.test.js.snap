// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetching source networks is fulfilled 1`] = `
Object {
  "errorSourceNetworks": null,
  "errorTargetNetworks": null,
  "isFetchingSourceNetworks": false,
  "isFetchingTargetNetworks": false,
  "isRejectedSourceNetworks": false,
  "isRejectedTargetNetworks": false,
  "sourceNetworks": Array [
    Object {
      "allow_promiscuous": null,
      "computed_allow_promiscuous": null,
      "computed_forged_transmits": null,
      "computed_mac_changes": null,
      "created_on": "2017-06-01T04:24:18Z",
      "forged_transmits": null,
      "href": "http://localhost:3000/api/lans/10000000000007",
      "id": "1-1",
      "mac_changes": null,
      "name": "VMWareCluster1-vm_network1",
      "parent_id": null,
      "switch_id": "1",
      "tag": "23",
      "uid_ems": "VMWareCluster1-vm_network1",
      "updated_on": "2017-06-01T04:24:18Z",
    },
  ],
  "targetNetworks": Array [],
}
`;

exports[`fetching source networks is pending 1`] = `
Object {
  "errorSourceNetworks": null,
  "errorTargetNetworks": null,
  "isFetchingSourceNetworks": true,
  "isFetchingTargetNetworks": false,
  "isRejectedSourceNetworks": false,
  "isRejectedTargetNetworks": false,
  "sourceNetworks": Array [],
  "targetNetworks": Array [],
}
`;

exports[`fetching source networks is rejected 1`] = `
Object {
  "errorSourceNetworks": "error",
  "errorTargetNetworks": null,
  "isFetchingSourceNetworks": false,
  "isFetchingTargetNetworks": false,
  "isRejectedSourceNetworks": true,
  "isRejectedTargetNetworks": false,
  "sourceNetworks": Array [],
  "targetNetworks": Array [],
}
`;

exports[`fetching target networks is fulfilled 1`] = `
Object {
  "errorSourceNetworks": null,
  "errorTargetNetworks": null,
  "isFetchingSourceNetworks": false,
  "isFetchingTargetNetworks": false,
  "isRejectedSourceNetworks": false,
  "isRejectedTargetNetworks": false,
  "sourceNetworks": Array [],
  "targetNetworks": Array [
    Object {
      "allow_promiscuous": null,
      "computed_allow_promiscuous": null,
      "computed_forged_transmits": null,
      "computed_mac_changes": null,
      "created_on": "2017-06-01T04:24:18Z",
      "forged_transmits": null,
      "href": "http://localhost:3000/api/lans/10000000000019",
      "id": "1-1",
      "mac_changes": null,
      "name": "RHV Cluster 1 - ovirtmgmt1",
      "parent_id": null,
      "switch_id": "10000000000000",
      "tag": null,
      "uid_ems": "00000000-0000-0000-0000-000000000000",
      "updated_on": "2017-06-01T04:24:18Z",
    },
  ],
}
`;

exports[`fetching target networks is pending 1`] = `
Object {
  "errorSourceNetworks": null,
  "errorTargetNetworks": null,
  "isFetchingSourceNetworks": false,
  "isFetchingTargetNetworks": true,
  "isRejectedSourceNetworks": false,
  "isRejectedTargetNetworks": false,
  "sourceNetworks": Array [],
  "targetNetworks": Array [],
}
`;

exports[`fetching target networks is rejected 1`] = `
Object {
  "errorSourceNetworks": null,
  "errorTargetNetworks": "error",
  "isFetchingSourceNetworks": false,
  "isFetchingTargetNetworks": false,
  "isRejectedSourceNetworks": false,
  "isRejectedTargetNetworks": true,
  "sourceNetworks": Array [],
  "targetNetworks": Array [],
}
`;

exports[`sets default state 1`] = `
Object {
  "errorSourceNetworks": null,
  "errorTargetNetworks": null,
  "isFetchingSourceNetworks": false,
  "isFetchingTargetNetworks": false,
  "isRejectedSourceNetworks": false,
  "isRejectedTargetNetworks": false,
  "sourceNetworks": Array [],
  "targetNetworks": Array [],
}
`;
